File: src/main/java/br/com/jiratorio/service/DueDateService.java
package br.com.jiratorio.service;

import br.com.jiratorio.domain.DueDateType;
import br.com.jiratorio.domain.embedded.DueDateHistory;
import br.com.jiratorio.domain.vo.changelog.JiraChangelogItem;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public interface DueDateService {

    List<DueDateHistory> extractDueDateHistory(String dueDateCF, List<JiraChangelogItem> changelogItems);

    Long calcDeviationOfEstimate(List<DueDateHistory> dueDateHistories, LocalDateTime endDate, DueDateType dueDateType,
                                 Boolean ignoreWeekend, List<LocalDate> holidays);

}

File: src/main/kotlin/br/com/jiratorio/service/DueDateService.kt
package br.com.jiratorio.service

import br.com.jiratorio.domain.jira.changelog.JiraChangelogItem
import br.com.jiratorio.domain.entity.embedded.DueDateHistory

interface DueDateService {

    fun extractDueDateHistory(dueDateCF: String, changelogItems: List<JiraChangelogItem>): List<DueDateHistory>

}
